[Core] Support token auth in ray_syncer

sampan-s-nayak opened the pull request:
## Description
This PR adds support for token-based authentication in the Ray bi-directional syncer, for both client and server sides. It also includes tests to verify the functionality.

Commit:
Add authentication token logic and related tests

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
Add gRPC service and server logic with auth integration tests

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
revert unneeded changs from src/ray/rpc/tests/BUILD.bazel

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
readd dependencies

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'master' into token_auth_1

Commit:
address comments + fix build

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'token_auth_1' into token_auth_2

Commit:
address comments

Signed-off-by: sampan <sampan@anyscale.com>

edoakes committed:
Merge branch 'master' into token_auth_2

Commit:
[Core] token auth support in bidi-syncer and pubsub rpc

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
add missing import

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
address comments

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
Merge remote-tracking branch 'upstream/token_auth_2' into token_auth_2

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'master' into token_auth_2

Commit:
fix ci

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
Merge remote-tracking branch 'upstream/token_auth_2' into token_auth_2

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix build.bazel and imports

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint issues

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'token_auth_2' into token_auth_3

Commit:
fix ray_syncer tests

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
add pub-sub test

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
[Core] Support token auth in ray Pub-Sub

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'token_auth_2.5' into token_auth_3

Commit:
get rid of getToken()

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
Merge remote-tracking branch 'upstream/token_auth_3' into token_auth_3

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
address comments

Signed-off-by: sampan <sampan@anyscale.com>

edoakes committed:
fix

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>

edoakes committed:
fix

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>

edoakes committed:
fix

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>

edoakes reviewed (approved):
one minor nit otherwise LGTM

sampan-s-nayak committed:
Merge branch 'token_auth_2.5' into token_auth_3

sampan-s-nayak commented on src/ray/common/ray_syncer/ray_syncer_server.cc:
```diff
@@ -35,13 +37,39 @@ RayServerBidiReactor::RayServerBidiReactor(
-      server_context_(server_context) {
+      server_context_(server_context),
+      auth_token_(auth_token) {
+  if (auth_token_.has_value() && !auth_token_->empty()) {
```
this is to handle the case where AuthToken is present but it is an empty string (auth loader should catch and throw an error in this scenario but added this extra check here just in case)

sampan-s-nayak committed:
Merge branch 'master' into token_auth_2.5

Commit:
[Core] Introduce new macros for user facing exceptions

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint issues

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
dont print stack trace for user errors

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
address comments

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'user_error_macro' into token_auth_2.5

Commit:
use RAY_USER_CHECK instead of RAY_CHECK + fix test

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
improve test

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'token_auth_2.5' into token_auth_3

Commit:
fix builds

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
lint

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'master' into user_error_macro

sampan-s-nayak committed:
Merge branch 'user_error_macro' into token_auth_2.5

Commit:
attempt to fix test

Signed-off-by: Sampan S Nayak <sampan@anyscale.com>

Commit:
attempt to fix test

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
attempt to fix test

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'token_auth_2.5' into token_auth_3

Commit:
fix test

Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
[Core] Support token auth in ray Pub-Sub  (#58186)

This PR adds token-based authentication support to the
PythonGcsSubscriber, which previously made direct gRPC calls via the
stub without auth. The rest of the pub-sub layer already uses the shared
gRPC infrastructure (GrpcServer, GrpcClient), which supports token
authentication.

---------

Signed-off-by: Sampan S Nayak <sampansnayak2@gmail.com>
Signed-off-by: sampan <sampan@anyscale.com>
Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Co-authored-by: sampan <sampan@anyscale.com>
Co-authored-by: Edward Oakes <ed.nmi.oakes@gmail.com>

sampan-s-nayak committed:
Merge branch 'user_error_macro' into token_auth_3

Signed-off-by: Sampan S Nayak <sampansnayak2@gmail.com>

edoakes commented on src/ray/common/tests/ray_syncer_test.cc:
```diff
@@ -983,6 +988,200 @@ TEST_F(SyncerReactorTest, TestReactorFailure) {
+
+  // Should connect successfully with matching token
+  client.Connect(channel);
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
```
why do we need the sleeps in these tests?

sampan-s-nayak replied:
the raysyncer code uses boost::asio for running things async, the sleep waits for these async calls to end.

Commit:
revert logging changes

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
address comments
Signed-off-by: sampan <sampan@anyscale.com>

sampan-s-nayak committed:
Merge branch 'token_auth_2.5' into token_auth_3

Signed-off-by: Sampan S Nayak <sampansnayak2@gmail.com>

Commit:
Merge remote-tracking branch 'upstream/token_auth_3' into token_auth_3

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
Merge remote-tracking branch 'upstream/master' into token_auth_3

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix build

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix lint

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix build

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
fix issue during merge conflict

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
revert unneeded changes

Signed-off-by: sampan <sampan@anyscale.com>

Commit:
address cursor comment

Signed-off-by: sampan <sampan@anyscale.com>
